app/controllers/plannings_controller.rb

```diff
@@ -32,15 +32,60 @@ def voxa_refine
     strategy = find_strategy_by_id_or_current
 
     unless strategy
+      Rails.logger.warn "PlanningsController: No strategy found for Voxa refinement (user: #{current_user.id})"
       redirect_to planning_path, alert: "No strategy found to refine." and return
     end
 
+    Rails.logger.info "PlanningsController: Starting Voxa refinement for strategy #{strategy.id} (user: #{current_user.id})"
+
     begin
       Creas::VoxaContentService.new(strategy_plan: strategy).call
-      redirect_to planning_path(plan_id: strategy.id), notice: "Content items refined successfully with Voxa!"
+      Rails.logger.info "PlanningsController: Voxa refinement started successfully for strategy #{strategy.id}"
+      redirect_to planning_path(plan_id: strategy.id), notice: "Content refinement has been started! Please come back to this page in a few minutes to see your refined content."
+    rescue StandardError => e
```

Comment:
Given that here we are in a controller, the error handling should be at the service level and let the controller to handle only the redirection logic.